spark/src/main/kotlin/com/adevinta/spark/res/SparkStringAnnotations.kt

```diff
@@ -0,0 +1,91 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.adevinta.spark.res
+
+import android.util.Log
+import androidx.compose.ui.text.SpanStyle
+import com.adevinta.spark.tokens.SparkColors
+import com.adevinta.spark.tokens.SparkTypography
+
+public interface SparkStringAnnotations<T : Any> {
+
+    public fun toSpanStyle(value: String, token: T): SpanStyle?
+
+    public object Colors : SparkStringAnnotations<SparkColors> {
+        public override fun toSpanStyle(value: String, token: SparkColors): SpanStyle = when (value) {
+            "primary" -> token.primary
+            "secondary" -> token.secondary
+            "success" -> token.success
+            "alert" -> token.alert
+            "error" -> token.error
+            "info" -> token.info
+            "neutral" -> token.neutral
+            else -> null
+        }?.let(::SpanStyle) ?: SpanStyle().also { _ ->
+            Log.d("StringResources", "Spark color annotation : $value is not supported")
+        }
+    }
+
+    public object Typography : SparkStringAnnotations<SparkTypography> {
+        public override fun toSpanStyle(value: String, token: SparkTypography): SpanStyle = when (value) {
+            "title1" -> token.title1
+            "title2" -> token.title2
+            "title3" -> token.title3
+            "bodyImportant" -> token.bodyImportant
+            "body" -> token.body
+            "smallImportant" -> token.smallImportant
+            "small" -> token.small
+            "extraSmallImportant" -> token.extraSmallImportant
+            "extraSmall" -> token.extraSmall
+            "button" -> token.button
+            else -> null
+        }?.toSpanStyle() ?: SpanStyle().also { _ ->
+            Log.d("StringResources", "Spark typography annotation : $value is not supported")
+        }
+    }
+
+    public companion object {
+        public fun toSpanStyle(
+            key: String,
+            value: String,
+            colors: SparkColors,
+            typography: SparkTypography,
+        ): SpanStyle {
+            return when (key.asSparkAnnotation()) {
+                Colors -> Colors.toSpanStyle(value, colors)
+                Typography -> Typography.toSpanStyle(value, typography)
+                else -> SpanStyle()
+            }
+        }
+
+        private fun String.asSparkAnnotation(): SparkStringAnnotations<*>? {
+            return when (this) {
+                "color" -> Colors
+                "typography" -> Typography
+                else -> null.also { _ ->
+                    Log.d("StringResources", "Annotation  $this is not supported by spark")
+                }
+            }
+        }
+    }
```

Comment:
IMO we don't need this extra layer of API as it does not bring any benefit vs directly calling

```kotlin
        public fun toSpanStyle(
            key: String,
            value: String,
            colors: SparkColors,
            typography: SparkTypography,
        ): SpanStyle {
            return when (key) {
                "color" -> Colors.toSpanStyle(value, colors)
                "typography" -> Typography.toSpanStyle(value, typography)
                else -> /* log */ SpanStyle()
            }
        }
```